settings.gradle

```diff
@@ -8,3 +8,4 @@
  */
 
 rootProject.name = 'plugin-template'
+startParameter.excludedTaskNames=["validatePluginZipPom"]
```

Comment:
Sorry, where this setting comes from?

Reply:
This is part of the plugin setup, usually all plugins managed build scripts or default build scripts run `assemble` task and `publish` tasks (specific to repo), example [job-scheduler](https://github.com/opensearch-project/job-scheduler/blob/main/scripts/build.sh#L70-L75), but not direct build task, if called we need to exclude this `validatePluginZipPom` as general `build` task will include this, if this task called it will fail as the POM validation with the error 

```
> Task :validatePluginZipPom FAILED
Execution optimizations have been disabled for task ':validatePluginZipPom' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/usr/share/opensearch/git/opensearch-plugin-template-java/build/distributions/rename-unspecified.pom'. Reason: Task ':validatePluginZipPom' uses this output of task ':generatePomFileForNebulaPublication' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.4.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
name is missing in [/usr/share/opensearch/git/opensearch-plugin-template-java/build/distributions/rename-unspecified.pom]
description is missing in [/usr/share/opensearch/git/opensearch-plugin-template-java/build/distributions/rename-unspecified.pom]
licenses is empty in [/usr/share/opensearch/git/opensearch-plugin-template-java/build/distributions/rename-unspecified.pom]
developers is empty in [/usr/share/opensearch/git/opensearch-plugin-template-java/build/distributions/rename-unspecified.pom]
```
Ideally we dont need to run this task as we define the POM values during actual publish.

Reply:
Just to add I have also added this point to the plugin usage [document](https://github.com/opensearch-project/opensearch-plugins/pull/144/files#diff-f1355fd54329147c830843ab133fed7716180c3242031487b861a2a3235d0078R250-R252) (PR under review)

Reply:
:+1:

Reply:
This still looks wrong to me, shouldn't we be fixing the POM by adding these fields?

Reply:
It was fixed after adding
```
allprojects {
    project.ext.licenseName = 'The Apache Software License, Version 2.0'
    project.ext.licenseUrl = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
    publishing {
        publications {
            all {
                pom.withXml { XmlProvider xml ->
                    Node node = xml.asNode()
                    node.appendNode('inceptionYear', '2022')
                    node.appendNode('name', 'rename')
                    node.appendNode('description', 'Custom plugin')
                    Node license = node.appendNode('licenses').appendNode('license')
                    license.appendNode('name', project.licenseName)
                    license.appendNode('url', project.licenseUrl)
                    Node developer = node.appendNode('developers').appendNode('developer')
                    developer.appendNode('name', 'OpenSearch')
                    developer.appendNode('url', 'https://github.com/opensearch-project/opensearch-plugin-template-java')
                }
            }
        }
    }
}
```
Then we dont need to exclude, I would say we can add the above template to the `build.gradle` and it's up-to the plugin team, if they want to exclude the the task `validatePluginZipPom` or add the POM fields in right way, I can update the info in the document as well, thoughts? @dblock

Reply:
excluding is better, rather than forcing user to add these POM fields as then it allow user to run `./gradlew build`, without any restrictions, @dblock @reta @bbarani @peterzhuamazon

Reply:
some plugins do not have all of these settings @dblock, but in our case to publish zips we dont need this task `validatePluginZipPom`.